velox/functions/lib/LikeFunctionUtils.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <string>
+
+namespace facebook::velox::functions {
+namespace likeutils {
+
+// This function checks if the string 'pattern' has wildcard characters; a
+// wildcard character could either be '_' or '%'. If wildcard characters are
+// detected in the string 'pattern', the function returns true; else it returns
+// false.
+bool checkWildcardInPattern(const std::string& pattern);
+
+// This function is called if the string 'pattern' has no wildcard characters; a
+// wildcard character could either be '_' or '%'. When there are no wildcard
+// characters in the pattern, the 'inputString' can be matched with the string
+// 'pattern' by comparing the characters at the same indices in 'inputString'
+// and 'pattern'. This character by character comparison is performed in this
+// function and if there is no mismatch till the end of both 'inputString' and
+// 'pattern' are reached, we return true.
+bool matchExactPattern(
+    const std::string& inputString,
+    const std::string& pattern);
+
+// This function checks if the string 'pattern' is an 'exact' pattern followed
+// by one or more occurences of the wildcard character, '%'; let us refer to
+// such patterns as prefix patterns. We define an 'exact' pattern as a string
```

Comment:
This statement can be rephrased

"This function checks if the string is a 'prefixPattern'. We define a prefixPattern as a 'exactPattern' (composed of fixed characters with no wildcards) followed by '%'."